### PR TITLE
App create api

### PR DESCRIPTION
**Pull request by manish-agrawal-ai**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test case A
- [ ] Test case B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have tested it for all user roles

### CODE REVIEW

File: api/restHandler/AppRestHandler.go [hunk 1]
```diff
@@ -0,0 +1,1962 @@
...
+	"go.uber.org/zap"
+	"gopkg.in/go-playground/validator.v9"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	APP_DELETE_FAILED_RESP     = "Devtron was unable to delete the app by itself, please try deleting the app manually."
```
Comment (pghildiyal) [inline api/restHandler/AppRestHandler.go:55]: App deletion failed, please try deleting from Devtron UI

File: api/restHandler/AppRestHandler.go [hunk 2]
```diff
@@ -0,0 +1,1962 @@
...
+		return
+	}
+	//rbac implementation ends here for app
+
+	handler.logger.Debugw("Getting app detail v2", "appId", appId)
+
+	//get/build app metadata starts
+	appMetadataResp, done := handler.validateAndBuildAppMetadata(w, appId)
+	if done {
+		return
```
Comment (pghildiyal) [inline api/restHandler/AppRestHandler.go]: this return as no response body, it should convey to user what  happened

File: api/restHandler/AppRestHandler.go [hunk 3]
```diff
@@ -0,0 +1,1962 @@
...
+	}
+	//get/build app metadata ends
+
+	//get/build git materials starts
+	gitMaterialsResp, done := handler.validateAndBuildAppGitMaterials(w, appId)
+	if done {
+		return
```
Comment (pghildiyal) [inline api/restHandler/AppRestHandler.go]: this return as no response body, it should convey to user what happened

File: api/restHandler/AppRestHandler.go [hunk 4]
```diff
@@ -0,0 +1,1962 @@
...
+	}
+	//get/build git materials ends
+
+	//get/build docker config starts
+	dockerConfig, done := handler.validateAndBuildDockerConfig(w, appId)
+	if done {
+		return
```
Comment (pghildiyal) [inline api/restHandler/AppRestHandler.go]: this return as no response body, it should convey to user what happened

File: api/restHandler/AppRestHandler.go [hunk 5]
```diff
@@ -0,0 +1,1962 @@
...
+	}
+	//get/build docker config ends
+
+	//get/build global deployment template starts
+	globalDeploymentTemplateResp, done := handler.validateAndBuildAppDeploymentTemplate(w, appId, 0)
+	if done {
+		return
```
Comment (pghildiyal) [inline api/restHandler/AppRestHandler.go]: this return as no response body, it should convey to user what happened

File: api/restHandler/CoreAppRestHandler.go [hunk 6]
```diff
@@ -0,0 +1,1962 @@
...
+		return
+	}
+	//creating blank app ends
+
+	//declaring appId for creating other components of app
+	appId := createBlankAppResp.Id
+
+	// rbac after create blank app
+	resourceObject := handler.enforcerUtil.GetAppRBACNameByAppId(appId)
+	if ok := handler.enforcer.Enforce(token, rbac.ResourceApplications, rbac.ActionCreate, resourceObject); !ok {
```
Comment (pghildiyal) [inline api/restHandler/CoreAppRestHandler.go:281]: unnecessary as this person has created the app so would have create access

File: api/restHandler/common/apiError.go [hunk 7]
```diff
@@ -59,6 +60,16 @@ func WriteJsonResp(w http.ResponseWriter, err error, respBody interface{}, statu
 			apiErr.UserMessage = err.Error()
 		}
 		response.Errors = []*util.ApiError{apiErr}
+	} else if multiErr, ok := err.(*multierror.Error); ok {
+		var errorsResp []*util.ApiError
+		for _, e := range multiErr.Errors {
+			errorResp := &util.ApiError{
+				UserMessage:     fmt.Sprint(e),
```
Comment (pghildiyal) [inline api/restHandler/common/apiError.go:67]: cant we use e.Error()

Comment (kartik-579) [reply api/restHandler/common/apiError.go]: we can, updated

**Review by pghildiyal — APPROVED** [review]

**Pull request merged**